cmake/Sol2.cmake

```diff
@@ -0,0 +1,43 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+# Using file(DOWNLOAD) instead of FetchContent due to this issue in cmake versions older than 3.18 https://gitlab.kitware.com/cmake/cmake/-/issues/20526
+
+set(SOL2_INCLUDE_DIR "${CMAKE_BINARY_DIR}/_deps/sol2/" CACHE STRING "" FORCE)
+if(NOT EXISTS "${SOL2_INCLUDE_DIR}/sol.hpp")
```

Comment:
it seems to me that other libraries use `FetchContent`

Reply:
The problem with `FetchContent` is that if we want to use it for single files that are not archives we have to use the `
DOWNLOAD_NO_EXTRACT` option. Unfortunately due to an issue it does not copy it to the expected directory and leaves it in an unreachable subdirectory. The issue is fixed in CMake 3.18 but we only require 3.16. More information about this issue can be found here: https://gitlab.kitware.com/cmake/cmake/-/issues/20526

Reply:
oh now I see there is a comment about it